surreal/http.py

```diff
@@ -0,0 +1,124 @@
+"""
+Copyright © SurrealDB Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import base64
+import logging
+from typing import Any
+from urllib.parse import quote
+
+from aiohttp import ClientSession
+
+from surreal.errors import SurrealException
+from surreal.utils import load_json
+
+from . import __version__
+
+_log = logging.getLogger(__name__)
+
+
+def encode_basic_auth(username: str, password: str):
+    username_password = f"{quote(username)}:{quote(password)}"
+
+    return f"Basic {base64.b64encode(username_password.encode()).decode()}"
```

Comment:
This and the resulting framework for sending auth is unnecessary, you can instead use `aiohttp.BasicAuth` and pass the resulting BasicAuth object to the session or request.